websites/N/Ngewibu/presence.ts

```diff
@@ -0,0 +1,26 @@
+const presence = new Presence({
+		clientId: "1104875968134397963",
+	}),
+	presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/ZmylEPM.png",
+		startTimestamp: Math.floor(Date.now() / 1000),
+	};
+
+presence.on("UpdateData", async () => {
+	if (document.location.pathname.startsWith("/play")) {
+		presenceData.details = "Watching anime";
+		presenceData.state = document
+			.querySelector("title")
+			.textContent.replace("Nonton ", "")
+			.replace(/- NgewibuTV/gi, "");
+		presenceData.buttons = [
+			{ label: "View Anime", url: document.location.href },
+		];
+	} else {
+		presenceData.details = "Viewing index";
+		presenceData.buttons = [
+			{ label: "Visit NgewibuTV", url: document.location.href },
+		];
+	}
+	presence.setActivity(presenceData);
+});
```

Comment:
Explanation:
```ts
const presence = new Presence({
		clientId: "1101840416866844672",
	}),
	browsingTimestamp = Math.floor(Date.now() / 1000); // Declaring the start timestamp
// All images used in a way for them to be read easily
enum Assets {
	Logo = "https://i.imgur.com/ZmylEPM.png",
	Play = "https://i.imgur.com/6s4WyWY.png",
	Pause = "https://i.imgur.com/PrYtpQb.png",
	Search = "https://i.imgur.com/08wjeL0.png",
}
presence.on("UpdateData", async () => {
	const presenceData: PresenceData = {
			largeImageKey: Assets.Logo,
			startTimestamp: browsingTimestamp,
		},
		{ href, pathname, hostname } = document.location,
		video = document.querySelector<HTMLVideoElement>("video"), // Get vid el
		search = document.querySelector<HTMLInputElement>("[id='search-input']"); // Get search el
	if (search?.value) { // if there's something in the search box then display it
		presenceData.details = "Searching for:";
		presenceData.state = search?.value;
		presenceData.smallImageKey = Assets.Search;
	} else if (pathname.startsWith("/play")) { //if on /play
		const thumbnail = document
			.querySelector('[class="img-fluid"]')
			?.getAttribute("src")
			?.replace(".webp", ".png");
		presenceData.details = document.querySelector(
			'[class="mb-1  text-truncate"]'
		)?.textContent; // Get anime name
		presenceData.state = document
			.querySelector('[class*="currentPlay bg-success"]')
			?.textContent.replace("E", "Episode"); // Get current episode number 
		presenceData.largeImageKey = thumbnail
			? `https://${hostname}${thumbnail}`
			: Assets.Logo; // If thumbnails display that otherwise the logo

		if (video) { // If there's a vid playing
	if (!video.paused) { [presenceData.endTimestamp, ] = presence.getTimestampsfromMedia(video)
			presenceData.smallImageKey = video.paused ? Assets.Pause : Assets.Play;
			presenceData.smallImageText = video.paused ? "Paused" : "Playing";
			presenceData.buttons = [{ label: "Watch Video", url: href }];
		} else presenceData.buttons = [{ label: "View Anime", url: href }];
	} else if (pathname === "/index") {
		presenceData.details = "Browsing all animes";
		presenceData.buttons = [{ label: "Browse All Animes", url: href }];
	} else if (pathname.includes("/index")) {
		presenceData.details = `Viewing ${pathname
			.split("/")[2]
			.toLowerCase()} animes`;
		presenceData.state = `Page ${
			document.querySelector('[class="page-item active"]')?.textContent
		}/${
			document.querySelector(
				'[class="pagination pagination-sm justify-content-center"]'
			).children[
				Number(
					document.querySelector(
						'[class="pagination pagination-sm justify-content-center"]'
					).children.length
				) - 2
			].textContent
		}`;
		presenceData.buttons = [{ label: "View Animes", url: href }];
	}
	if (presenceData.details) presence.setActivity(presenceData);
	else presence.setActivity();
});
```